### PR TITLE
[gemmlowp] create a new port

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?  

Create a new port `gemmlowp` with `find_package` support.

```cmake
find_package(gemmlowp CONFIG REQUIRED)
target_link_libraries(main PRIVATE gemmlowp::gemmlowp gemmlowp::eight_bit_int_gemm)
```

(Changes are moved to https://github.com/google/gemmlowp/pull/207)

This project uses `SHARED` for non-Windows target. Please check [contrib/CMakeLists.txt](https://github.com/google/gemmlowp/blob/master/contrib/CMakeLists.txt).

`gemmlowp` is a `INTERFACE` target to include directory `${INCLUDEDIR}/gemmlowp`.
The actual library is `eight_bit_int_gemm`.

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

* Targets `windows` and `osx` triplets

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

* Build Techniques
    * [x] Prefer using CMake
* Versioning
    * [x] `version-date`
* Do not build tests/docs/examples

### CODE REVIEW

File: ports/gemmlowp/support-install.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
++        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
++        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
++install(EXPORT  gemmlowp-config
++        NAMESPACE unofficial::
++        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/gemmlowp)
++        
++return()
+ # Benchmarks
+ add_executable(benchmark
+     "${gemmlowp_src}/test/benchmark.cc" ${gemmlowp_test_headers})
```
Comment (NancyLi1013) [inline ports/gemmlowp/support-install.patch:38]: Could you please submit the changes to upstream?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/gemmlowp/support-install.patch (hunk 1 above)
Comment (luncliff) [reply ports/gemmlowp/support-install.patch:38]: Sure. Let me ask them. This part will be used for vcpkg's CI until it's ready.

**Comment by luncliff** [comment]
Hope https://github.com/google/gemmlowp/pull/207 goes well.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for adding this new port. @luncliff

**Comment by BillyONeal** [comment]
Thanks for the new port and for working with upstream to make them better cmake citizens!

**Pull request merged**